build.sbt

```diff
@@ -96,7 +95,7 @@ lazy val plugin =
       libraryDependencies ++= Seq(
         "org.scala-lang.modules" %% "scala-xml" % "1.3.0",
         "org.scalatest" %% "scalatest" % scalatestVersion % Test,
-        "org.scala-lang" % "scala-compiler" % scalaVersion.value % Compile
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % Provided
```

Comment:
So I'm not 100% sure on this. Tonight I took some time and looked at other various compiler plugins. Most of them do actually just include it normally as a dependency, but there are a couple that do use `Provided`. From testing locally this seems to be fine, and it does seem to make sense that this can be provided, but I'd also like to check my logic on this before actually committing it.

Reply:
Provided is totally fine, as a Scala compiler plugin will be executed by the compile and thus should also be run in a classpath that already comes with it. Also, it has the effect, that it will not widen the transitive set of dependencies, but that should not be our main focus.